### PR TITLE
Update dependencies for the main ttrpc project

### PR DESCRIPTION
**Pull request by jsturtevant**
Updates the dependencies for the main ttrpc project.  This is in preparation for a release to include the Windows files in support for projects like the rust shim in https://github.com/containerd/rust-extensions/pull/139

### CODE REVIEW

**Comment by jsturtevant** [comment]
The nix dependency is failing due to:

```
error: use of deprecated enum `nix::sys::socket::SockAddr`: use SockaddrLike or SockaddrStorage instead
   --> src/common.rs:107:62
```

Is this something that should be addressed or should we move back to an older version and address later?

**Comment by Tim-Zhang** [comment]
> The nix dependency is failing due to:
> 
> ```
> error: use of deprecated enum `nix::sys::socket::SockAddr`: use SockaddrLike or SockaddrStorage instead
>    --> src/common.rs:107:62
> ```
> 
> Is this something that should be addressed or should we move back to an older version and address later?

I think we could adapt new nix API by using `SockaddrLike` instead of `SockAddr`

**Comment by jsturtevant** [comment]
@Tim-Zhang I've update to use those API's.  

Another option instead of boxing the trait would be to refactor some of these functions where `make_socket` are used. I don't know enough about the impact of Boxing to know if a bigger refactor of those functions makes sense or this is good enough.

File: src/common.rs [hunk 1]
```diff
@@ -80,13 +80,16 @@ pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::SOCK_CLOEXEC;
 pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::empty();
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
-fn make_addr(domain: Domain, sockaddr: &str) -> Result<UnixAddr> {
+fn make_addr(domain: Domain, sockaddr: &str) -> Result<Box<dyn SockaddrLike>> {
```
Comment (Tim-Zhang) [inline src/common.rs:83]: I think this method could keep unchanged. update return value of method make_addr is enough.

Comment (jsturtevant) [reply src/common.rs]: good call, updated!

**Review by Tim-Zhang — APPROVED** [review]
LGTM, thanks @jsturtevant

**Review by lifupan — APPROVED** [review]

**Pull request merged**